### PR TITLE
Sharing: Clarify Connection Prompt

### LINKED ISSUE #24094: Sharing: Clarify wording of oAuth post-connection dialog
#### Steps to reproduce
1. Starting at URL: https://wordpress.com/sharing/{site}
2. Connect to a new Sharing service (Twitter, Facebook, etc)
3. The dialog that comes up reads:

screen-shot-2018-03-26-at-02-09-03

This is somewhat confusing in that it might sound like all existing or old posts will be shared, and doesn't offer a clear next step.

We should reword it to be a bit more straightforward/friendly (we can drop constructions like "Note that...") and to make it clear what will happen if the user goes ahead with the connection, and what they still need to do.

### PR DESCRIPTION
**Pull request by torres126**
#### Changes proposed in this Pull Request

Replace the current text with the one used in the original issue, in favour of clarity. 

#### Testing instructions

Follow the instructions in the original issue, but note the difference in the prompt which appears. 

**Current:**

> Confirm this is the account you would like to authorize. Note that your posts will be automatically shared to this account.

**Proposed:**

> Is this the account you'd like to connect? All your new blog posts will be automatically shared to this account. You'll be able to change this option in the editor sidebar when you're writing a post.

(cc @codebykat & @andfinally) 

Fixes #24094

### CODE REVIEW

File: client/my-sites/sharing/connections/account-dialog.jsx [hunk 1]
```diff
@@ -176,7 +176,8 @@ class AccountDialog extends Component {
 			// If a single account is available, show a simple confirmation
 			// prompt to ask the user to confirm their connection.
 			return this.props.translate(
-				'Confirm this is the account you would like to authorize. Note that your posts will be automatically shared to this account.',
+				"Is this the account you'd like to connect? All your new blog posts will be automatically shared to this account. " +
```
Comment (a8ci18n) [inline client/my-sites/sharing/connections/account-dialog.jsx]: :information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

Comment (andfinally) [reply client/my-sites/sharing/connections/account-dialog.jsx]: A `context` message like the one on line 182 is normally useful for a short translated string whose meaning might be different in different places: it gives translators who may not see the surrounding code some information about where the string is used. A `comment` explains how the string is used, or what it's trying to say. In the light of the i18n bot's suggestion, we should change the `context` here to a `comment` like `Sharing: asks the user to confirm if they want to share future posts to a connected social media account.`. 

The same `context` is used on the other string with this phrasing, on line 191, so we should change it there too.

**Comment by andfinally** [comment]
Many thanks for taking this on @torres126, your PR description is very helpful! ❤️I added a couple of notes.

I notice there's another place in this dialog, around line 190, where we have a very similar message: `Select the account you wish to authorize. Note that your posts will be shared to the selected account automatically.` We should probably update that wording too. I suggest:

```
Select the account you'd like to connect. All your new blog posts will be automatically shared to this account. You'll be able to change this option in the editor sidebar when you're writing a post.
```

One of the automated tests is failing with a message like

```
line 214, col 9, Error - className should follow CSS namespace guidelines (expected connections__ prefix) (wpcalypso/jsx-classname-namespace)
```

This is an ESLint test which enforces the format for `className` attributes on React components. The rule expects that the first part of a component's CSS class will be the same as the folder the component is in. So the linter is expecting the class to be, for example, `connect__authorizing-service` instead of `account-dialog__authorizing-service`.

This component was made before that particular test was added to Calypso, and it's part of a fairly complex organization of parts. Rather than trying to rearrange the whole set of components I think it will be OK for us to override the test in this particular case. We can do this by adding comments to the `render` function which tell the linter not to worry about that rule. So before line 208 we'd add

```
/*eslint-disable wpcalypso/jsx-classname-namespace */
```

and after line 225 we'd turn the rule on again with:

```
/*eslint-enable wpcalypso/jsx-classname-namespace */
```

You should see some other examples of these comments elsewhere in the code.

This apparently straightforward change turned out to have a few complications, but hopefully they're fairly easy to iron out!

**Comment by torres126** [comment]
Thanks for the detailed review and help @andfinally, added a commit to try and address them!

File: client/my-sites/sharing/connections/account-dialog.jsx [hunk 2]
```diff
@@ -187,9 +187,9 @@ class AccountDialog extends Component {
 		// single Keyring connection, and the user must choose which
 		// account to connect.
 		return this.props.translate(
-			'Select the account you wish to authorize. Note that your posts will be shared to the selected account automatically.',
+			"Select the account you'd like to connect. All your new blog posts will be automatically shared to this account. You'll be able to change this option in the editor sidebar when you're writing a post.',
```
Comment (andfinally) [inline client/my-sites/sharing/connections/account-dialog.jsx]: The linter's complaining about an unterminated string here - this is likely because we've ended the string with a `'` instead of a `"`.

**Comment by andfinally** [comment]
Thanks @torres126, I think we're nearly there! There's just an issue with a quote mark on line 190.

File: client/my-sites/sharing/connections/account-dialog.jsx (hunk 2 above)
Comment (torres126) [reply client/my-sites/sharing/connections/account-dialog.jsx]: Oops, sorry, fixed!

Comment (torres126) [reply client/my-sites/sharing/connections/account-dialog.jsx]: Hmm, it looks like the tests are still failing. I don't _think_ it's related to this change though?

Comment (andfinally) [reply client/my-sites/sharing/connections/account-dialog.jsx]: You can often get a clue about what the problem is if you click on the `Details` link next to the failing group of tests. It shows the linter's complaining about the CSS classes again, this time in the `getConnectedAccountsContent` method. This is the only other place where `className` attributes are set, so if we add some another pair of ESLint comments that should fix them.

Looking at other places where these comments are used, they're commonly placed before and after the `return( ... );` statement, so if you add them above line 147 and below line 166, and move the others to above 206 and below 227, we should be good - fingers crossed!

Comment (torres126) [reply client/my-sites/sharing/connections/account-dialog.jsx]: Looks like the linter is happy now. :smile:

Thanks for the help with this!

Comment (andfinally) [reply client/my-sites/sharing/connections/account-dialog.jsx]: Great stuff! 🎉I'll give this another run-through on my local environment tomorrow, and hopefully I can get this merged for you, thanks!

**Comment by andfinally** [comment]
Hi @torres126, I've checked your branch out and made a few small changes:

* Moved the linter comments around a bit - one of the functions still had comments within the `render` function, and the other function had them incorrectly nested.
* Rebased the branch on master, so it's up to date with the latest changes to master.
* Squashed your commits into one commit, to keep the commit history cleaner.

Is it OK if I do a "force push" to your fork? This will overwrite your local history with these changes.

**Comment by torres126** [comment]
Go for it, thanks! :+1:
